### DISCUSSION TITLE
pdfjs-dist failed to load its dependency - Module not found: Package path . is not exported from package /node_modules/pdfjs-dist/node_modules/path2d-polyfill

### DISCUSSION THREAD

**Discussion by hpardess** [mozilla/pdf.js#17807] [Q&A]
Attach (recommended) or Link to PDF file here:
<img width="584" alt="Screenshot 2024-03-17 at 1 19 09 PM" src="https://github.com/mozilla/pdf.js/assets/6196491/8506ff6d-166f-41ad-8d3d-10725460bba0">

Here is the error message:
```
 ⨯ ./node_modules/pdfjs-dist/build/pdf.min.mjs:21:262714
Module not found: Package path . is not exported from package /project/node_modules/pdfjs-dist/node_modules/path2d-polyfill (see exports field in /project/node_modules/pdfjs-dist/node_modules/path2d-polyfill/package.json)

https://nextjs.org/docs/messages/module-not-found
```

Configuration:
```
Operating System:
  Platform: darwin
  Arch: arm64
  Version: Darwin Kernel Version 23.2.0: Wed Nov 15 21:55:06 PST 2023
Binaries:
  Node: 20.9.0
  npm: 10.1.0
  Yarn: N/A
  pnpm: N/A
Relevant Packages:
  next: 14.1.0
  eslint-config-next: 14.1.3
  react: 18.2.0
  react-dom: 18.2.0
  typescript: 5.4.2
Next.js Config:
  output: N/A
```

Steps to reproduce the problem:
https://codesandbox.io/p/sandbox/pdf-viewer-v52xpl

pdfjs-dist failed to load its dependency while using in Nextjs.

https://github.com/vercel/next.js/issues/63385

**Comment by lasseborly** [comment]
Using pdf.mjs instead of the pdfjs.min.mjs [works for me](https://codesandbox.io/p/devbox/pdf-viewer-v52xpl?file=%2Fcomponents%2FTestPage.tsx%3A4%2C1&layout=%257B%2522sidebarPanel%2522%253A%2522EXPLORER%2522%252C%2522rootPanelGroup%2522%253A%257B%2522direction%2522%253A%2522horizontal%2522%252C%2522contentType%2522%253A%2522UNKNOWN%2522%252C%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522id%2522%253A%2522ROOT_LAYOUT%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522UNKNOWN%2522%252C%2522direction%2522%253A%2522vertical%2522%252C%2522id%2522%253A%2522clu30bm8800083b6f1stpsdqo%2522%252C%2522sizes%2522%253A%255B70%252C30%255D%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522EDITOR%2522%252C%2522direction%2522%253A%2522horizontal%2522%252C%2522id%2522%253A%2522EDITOR%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522EDITOR%2522%252C%2522id%2522%253A%2522clu30bm8800023b6fhfwhg9da%2522%257D%255D%257D%252C%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522SHELLS%2522%252C%2522direction%2522%253A%2522horizontal%2522%252C%2522id%2522%253A%2522SHELLS%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522SHELLS%2522%252C%2522id%2522%253A%2522clu30bm8800053b6f9jwcvwfs%2522%257D%255D%252C%2522sizes%2522%253A%255B100%255D%257D%255D%257D%252C%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522DEVTOOLS%2522%252C%2522direction%2522%253A%2522vertical%2522%252C%2522id%2522%253A%2522DEVTOOLS%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522DEVTOOLS%2522%252C%2522id%2522%253A%2522clu30bm8800073b6fsqedu7zp%2522%257D%255D%252C%2522sizes%2522%253A%255B100%255D%257D%255D%252C%2522sizes%2522%253A%255B50%252C50%255D%257D%252C%2522tabbedPanels%2522%253A%257B%2522clu30bm8800023b6fhfwhg9da%2522%253A%257B%2522id%2522%253A%2522clu30bm8800023b6fhfwhg9da%2522%252C%2522tabs%2522%253A%255B%257B%2522id%2522%253A%2522clu30dn8o006f3b6f1ztg1i58%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522type%2522%253A%2522FILE%2522%252C%2522initialSelections%2522%253A%255B%257B%2522startLineNumber%2522%253A4%252C%2522startColumn%2522%253A1%252C%2522endLineNumber%2522%253A4%252C%2522endColumn%2522%253A1%257D%255D%252C%2522filepath%2522%253A%2522%252Fcomponents%252FTestPage.tsx%2522%252C%2522state%2522%253A%2522IDLE%2522%257D%255D%252C%2522activeTabId%2522%253A%2522clu30dn8o006f3b6f1ztg1i58%2522%257D%252C%2522clu30bm8800073b6fsqedu7zp%2522%253A%257B%2522tabs%2522%253A%255B%257B%2522id%2522%253A%2522clu30bm8800063b6f0otad45i%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522type%2522%253A%2522TASK_PORT%2522%252C%2522taskId%2522%253A%2522dev%2522%252C%2522port%2522%253A3000%252C%2522path%2522%253A%2522%252F%2522%257D%255D%252C%2522id%2522%253A%2522clu30bm8800073b6fsqedu7zp%2522%252C%2522activeTabId%2522%253A%2522clu30bm8800063b6f0otad45i%2522%257D%252C%2522clu30bm8800053b6f9jwcvwfs%2522%253A%257B%2522tabs%2522%253A%255B%257B%2522id%2522%253A%2522clu30bm8800033b6fh8k44ljk%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522type%2522%253A%2522TASK_LOG%2522%252C%2522taskId%2522%253A%2522dev%2522%257D%252C%257B%2522id%2522%253A%2522clu30bm8800043b6fhng4pvah%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522type%2522%253A%2522TERMINAL%2522%252C%2522shellId%2522%253A%2522cltvyfl3d01e5dbgeddh01ej4%2522%257D%255D%252C%2522id%2522%253A%2522clu30bm8800053b6f9jwcvwfs%2522%252C%2522activeTabId%2522%253A%2522clu30bm8800033b6fh8k44ljk%2522%257D%257D%252C%2522showDevtools%2522%253Atrue%252C%2522showShells%2522%253Atrue%252C%2522showSidebar%2522%253Atrue%252C%2522sidebarPanelSize%2522%253A15%257D).

Your issue then becomes Next.js being complicated and opinionated about how it loads something complicated like pdf.js.

I would suggest using the react-pdf abstraction instead, since they have specific Next.js guidance.
https://react-pdf.org/compatibility

**Comment by mmo80** [comment]
You can get it working with `pdfjs-dist ^4.0.379` (pdf.js) and `Next.js 14.1.4` with api route but same code as a Server Action throws error for some reason.

Check out the code [here](https://github.com/mmo80/pdfja-nextjs-bug) that showcases a api route and a server action
